streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBVersionedStore.java

```diff
@@ -0,0 +1,792 @@
+                        expiredRecordSensor.get().record(1.0d, ProcessorContextUtils.currentSystemTime(context));
+                        LOG.warn("Skipping record for expired put.");
+                    }
+                    return;
+                }
+
+                final byte[] segmentValue = segment.get(key);
+                if (segmentValue == null) {
+                    // in this special case where the latest record version (for a particular key)
+                    // is a tombstone, and the segment that the tombstone belongs in contains no
+                    // record versions for this key, create a new "degenerate" segment with the
+                    // tombstone's timestamp as both validFrom and validTo timestamps for the segment
+                    segment.put(key, segmentValueSchema.newSegmentValueWithRecord(null, timestamp, timestamp).serialize());
+                } else {
+                    // insert as latest, since foundTs = sentinel means nothing later exists
+                    if (segmentValueSchema.getNextTimestamp(segmentValue) == timestamp) {
+                        // next timestamp equal to put() timestamp already represents a tombstone,
+                        // so no additional insertion is needed in this case
+                        return;
+                    }
+                    final SegmentValue sv = segmentValueSchema.deserialize(segmentValue);
+                    sv.insertAsLatest(
+                        segmentValueSchema.getNextTimestamp(segmentValue),
+                        timestamp,
+                        null
+                    );
+                    segment.put(key, sv.serialize());
+                }
+            }
+        } else {
+            // insert into segment corresponding to foundTs, as foundTs represents the validTo
+            // timestamp of the current put.
+            // the new record is either the earliest or the latest in this segment, depending on the
+            // circumstances of the fall-through. (it cannot belong in the middle because otherwise
+            // putSegments() above would have identified a segment for which
+            // minTimestamp <= timestamp < nextTimestamp, and putSegments would've completed the
+            // put procedure without reaching this fall-through case.)
+            final T segment = versionedStoreClient.getOrCreateSegmentIfLive(
+                versionedStoreClient.segmentIdForTimestamp(foundTs), context, observedStreamTime);
+            if (segment == null) {
+                if (expiredRecordSensor.isPresent()) {
+                    expiredRecordSensor.get().record(1.0d, ProcessorContextUtils.currentSystemTime(context));
+                    LOG.warn("Skipping record for expired put.");
+                }
+                return;
+            }
+
+            final byte[] segmentValue = segment.get(key);
+            if (segmentValue == null) {
+                if (value != null) {
```

Comment:
Why do we need this `if`? Seems we can just do
```
segment.put(
  key,
  segmentValueSchema.newSegmentValueWithRecord(value, timestamp, foundTs).serialize()
);
```

If `value == null`, we still pass `null` as first argument.

Reply:
You're right! This is leftover from a previous version of the code. I will simplify it.